### PR TITLE
Handle democracy proposal cancel

### PR DESCRIPTION
**Pull request by markdavison**
- Add extrinsic handler for `Democracy:clear_public_proposals`
- Fix an issue with types that subsquid was choking on
- Add handling for fetching proposal_hash from BatchAll extrinsic

### CODE REVIEW

File: prawns/governance/src/handlers/democracy.Proposed.event.ts [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+import { SubstrateDemocracyProposal, SubstrateNetwork } from '../model';
 import { getDemocracyProposedEvent } from './typeGetters/getDemocracyProposedEvent';
 
+const getProposalHash = (ctx: EventHandlerContext, network: SubstrateNetwork) => {
+
+  const args = ctx.event!.extrinsic!.args;
+
+  if (ctx.event.extrinsic?.method === 'batchAll') {
+    const calls = args[0].value as { args: {[key: string]: any} }[];
+    const batchProposeArgs = calls.find(call => call.args && call.args.proposal_hash);
```
Comment (bdmason) [inline prawns/governance/src/handlers/democracy.Proposed.event.ts:12]: Is it possible to batch 2 of these extrinsic types? If so an event will be emitted for each, and the 1st extrinsic proposal hash will be applied in all the event handlers

File: prawns/governance/src/processors/khalaProcessor.ts [hunk 2]
```diff
@@ -83,6 +84,12 @@ processor.addEventHandler(
   'democracy.Started',
   democracyStartedEventHandler(network)
 );
-
+processor.addExtrinsicHandler(
+  'democracy.clear_public_proposals',
+  {
+    triggerEvents: ['treasury.Deposit'] // For some reason this extrinsic does not have a 'system.ExtrinsicSuccess' event that Subsquid looks for to trigger the handler
```
Comment (bdmason) [inline prawns/governance/src/processors/khalaProcessor.ts:90]: good find!

**Review by bdmason — APPROVED** [review]
Other than the 1 query about batch all this looks good, please look at it before merging

File: prawns/governance/src/handlers/democracy.Proposed.event.ts (hunk 1 above)
Comment (markdavison) [reply prawns/governance/src/handlers/democracy.Proposed.event.ts:12]: It's a good point. It's not been used before.. but I guess it might be possible?

Comment (markdavison) [reply prawns/governance/src/handlers/democracy.Proposed.event.ts]: Added some defensive code so if it ever happens it will error.

**Pull request merged**